zcash_primitives/src/transaction/components/transparent/builder.rs

```diff
@@ -188,6 +188,32 @@ impl TxIn<Unauthorized> {
     }
 }
 
```

Comment:
On lines 93..107 there is an implementation of Hash160, but:
* it's incorrectly called RIPEMD-160 in a comment;
* it uses the deprecated `ripemd160` crate, and [I think we want to switch to the `ripemd` crate](https://github.com/zcash/librustzcash/pull/341/files#r784282022);
* it's duplicated with the implementation in [#341](https://github.com/zcash/librustzcash/pull/341/files#r784325108).

I suggest pulling this out to a public helper method `hash160` in `transactions::components::transparent`, and calling it from `zcash_client_backend` (so that the choice of RIPEMD-160 implementation would be made in `zcash_primitives`).

Reply:
Filed #482.